### PR TITLE
add a walkthrough for setting up and trying the server on minikube

### PR DESCRIPTION
**Pull request by maleck13**
This PR addresses issue https://github.com/kubernetes/sample-apiserver/issues/8

### CODE REVIEW

**Comment by maleck13** [comment]
@sttts @deads2k  is this the right location for a docs PR or should this also be made against the core kubernetes repo

File: docs/minikube-walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+```
+cd $GOPATH/src
+mkdir -p github.com/kubernetes
+cd github.com/kubernetes
+git clone https://github.com/kubernetes/sample-apiserver.git
+``` 
+
+## Rename the package imports
+
+After cloning from the repo, the imports will be referencing ```k8s.io/sample-apiserver``` these need to be changed to ```github.com/kubernetes/sample-apiserver```. Recommend you do this with your favourite editor and a find all and replace.
```
Comment (sttts) [inline docs/minikube-walkthrough.md]: This shouldn't be necessary if we describe how to check it out into $GOPATH/src/k8s.io/sample-apiserver.

File: docs/minikube-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        - --requestheader-username-headers=X-Remote-User
+        - --requestheader-group-headers=X-Remote-Group
+        - --requestheader-extra-headers-prefix=X-Remote-Extra        
+```
+
+You need to change the ```imagePullPolicy``` to ```Always``` or ```IfNotPresent```.
+
+You also need to change the image from ```kube-sample-apiserver:latest``` to ```<YOUR_DOCKERHUB_USER>/kube-sample-apiserver:latest```
+
+Save this file and we are then ready to deploy and try out the sample apiserver.
```
Comment (sttts) [inline docs/minikube-walkthrough.md]: I wonder whether we shouldn't deliver artifacts for this use-case and avoid instructions like this to modify files. Would make it much easier.

**Comment by sttts** [comment]
> is this the right location for a docs PR

I just checked: the README.md is already authorative here in this repo. So I am fine with putting the docs here as well.

File: docs/minikube-walkthrough.md (hunk 1 above)
Comment (maleck13) [reply docs/minikube-walkthrough.md]: good point I will make that change

File: docs/minikube-walkthrough.md (hunk 2 above)
Comment (maleck13) [reply docs/minikube-walkthrough.md]: Not sure I am following. Do you mean provide a image somewhere like dockerhub? That would be useful, although I imagine most looking at this repo are thinking about developing their own apiserver and are using this as a template to get started. So they will want to create their own images. At least I know, from my perspective, I wanted to know how to get setup, build and deploy. I then immediately wanted to make changes and add my own resource etc to see how that worked.

Comment (sttts) [reply docs/minikube-walkthrough.md]: The DOCKERHUB_USER must be replaced I agree. I meant the other changes, especially the `--requestheader-*` lines. I am proposing to either change the ones we have or if that conflicts with another use of them, let's create a different set of artifacts.

@deads2k can you comment where we use these artifacts now? Do we at all?

Comment (maleck13) [reply docs/minikube-walkthrough.md]: @deads2k @sttts Yes agree, it would be better if we could add the required configuration to the existing artifacts

Comment (deads2k) [reply docs/minikube-walkthrough.md]: > @deads2k can you comment where we use these artifacts now? Do we at all?

They can be auto-detected in the cluster if the user is bound with a role like this: https://github.com/openshift/kubernetes-namespace-reservation/blob/master/artifacts/install/rbac-template.yaml#L38-L50

Comment (maleck13) [reply docs/minikube-walkthrough.md]: Yes from the configmap. But the issue (at least at the time of writing the walkthrough) was these setting were not present in minikube.

**Comment by maleck13** [comment]
@sttts updated based using the latest minikube

**Pull request merged**